### DISCUSSION TITLE
WalletConnect Desktop wallet list is a total mess and needs fixed ASAP!

### DISCUSSION THREAD

**Discussion by Crytogenz** [WalletConnect/walletconnect-monorepo#2693] [Developer Support]
The wallet list on desktop for WalletConnect is in a right state at the moment, it mostly doesn't contain desktop wallets anymore since an update a few days ago and a few of the top desktop wallets have been misplaced. **THIS DESPERATLY NEEDS SOLVED ASAP!** 

Here is the current order of wallets on the desktop WalletConnect, as well as those that need removed:
```
- Zerion = Remove
- Bitkeep = Remove
- Exodus = Remove
- Ledger = Perfect, but it needs placed back to position 1 as it has been since WalletConnect launch as now its position 4 on both walletconnect v1 and v2
- Fireblocks = Remove
- Ambire = Okay
- Now Wallet = Okay
- AT.Wallet = Remove
- Verso = Remove
- Broearn = Remove
- Certhis = Remove
- Cypherrock = TODO

- Infinity Wallet = Perfect, but it needs placed back to position 2-3 as it has been since WalletConnect launch as now its position 13 (page 2) on walletconnect v1 and position 11 on walletconnect v2.
- Vision = Remove
- ApolloX = Remove
- Sequence = Okay
- UniPass = Okay
- Marble = Does not work
- Crossmint = Does not work
- Pier = Remove
- Paper = Does not work
- RICE Wallet = Remove
- Bobablocks = Remove
- KryptoGO = Remove

- Wallet3 = Remove
- PREMA = Remove
- OneKey = Unsure
- SecuX = Remove
- Venly = Does not work
- Bitski = Unable to test
- SafeMoon = Remove
```

### Justification for each platform:
Below I have provided a justification of the reason why projects need removed, don't work or why position needs fixed.

**Zerion (Position 1)** = `Needs removed as DApp`
1- They don't have a desktop wallet only a web3 DApp that is not a wallet but requires a wallet to be used to connect to it. They only have a mobile wallet which supports WalletConnect, so would really like to know why a DApp is listed as position 1 desktop wallet when its not a desktop wallet?

2- I noticed that the wallet that use to be in Zerion position "Ledger" has now been removed to position 4, why was this?

**Bitkeep (Position 2)** = `Needs removed`
1- Its has a browser extension wallet, but it only supports WalletConnect on mobile. So why is it listed on the Desktop list if it can't be used on Desktop?

2- I noticed that the wallet that use to be in Bitkeep position "Infinity Wallet" has now been removed from page 1 position 2-3, even though it was one of the only desktop wallets on the list that actually worked other than ledger, so why was it pushed off the list? Also Bitkeep a few weeks ago was on page 3 and now its page 1 position 2-3 which is strange.

**Exodus (Position 3)** = `Needs removed`
It has a desktop wallet, but it doesn't support WalletConnect. You can only use WalletConnect on its mobile wallet, so why is its Desktop wallet on the Desktop list if it does not support WalletConnect? 

Also wondering how this jumped from the last page to the first page at the top while other actual working desktop wallets that have supported WalletConnect as long as I can remember such as Ledger, Infinity Wallet and etc.. were either removed or moved down the list?

**Ledger (Position 4)** = `PERFECT, needs placed back to its original position on the list` 
The ledger wallet has a desktop version and works perfectly fine on Desktop.

This wallet use to be positioned between position 1-2, however it was recently in the last week removed from its position and now on position 4, I am trying to understand why you removed one of the few actual true desktop wallets to support a bunch of none desktop wallets? 

**Fireblocks (Position 5)** = `Needs removed`
This is a centralized wallet it would seem as it requires email login, I was unable to even test WalletConnect as their website does not even make it possible to login. I am trying to understand why a centralized wallet would seem to be listed on WalletConnect when it doesnt work?

**Ambire (Position 6)** = `Okay`
They have a website based wallet which works fine with WalletConnect.

**Now Wallet (Position 7)** = `Okay`
They have a desktop wallet which works fine with WalletConnect.

**Broearn (Position 8)** = `Needs removed`
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet yet, it says on the website coming soon!

Website to check: https://www.broearn.com/download/android 

**AT.Wallet (Position 9)** = `Needs removed`
This is a mobile only wallet, with no website, 100 downloads on mobile, so why is it on the desktop wallet list?

**Certhis (Position 10)** = `Needs removed as DApp`
This is a DApp which requires a user to use another wallet to connect with it via WalletConnect, its not a wallet so why is it even listed on the desktop wallet list?

**Verso (Position 11)** = `Needs removed`
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet yet, it says on the website coming soon!

Website to check: https://get-verso.com/

**Cypherock (Position 12)** = `TODO`
TODO as need hardware wallet to test 

**Infinity Wallet (Position 13)** = `PERFECT, needs placed back to its original position on the list` 
The Infinity wallet has a desktop version and works perfectly fine on Desktop. This was the only desktop wallet along with ledger that I remember supporting WalletConnect from the start (when it was only those 2 wallets that existed) and like ledger it is one of few that actually works perfectly.

This wallet use to be positioned page 1 between position 2-3, however it was recently in the last week removed from page 1 rank 3 and now placed on page 2, I am trying to understand why you removed one of the few actual true desktop wallets to support a bunch of none desktop wallets? 

**Vision (Position 14)** = `Needs removed`
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://www.vision-crypto.com/

**ApolloX (Position 15)** = ``Needs removed as DApp``
This is a DApp which requires a user to use another wallet to connect with it via WalletConnect, its not a wallet so why is it even listed on the desktop wallet list?

**Sequence (Position 16)** = `Okay`
This is a very very basic website based wallet, but it supports WalletConnect. 

**Unipass(Position 17)** = `Okay`
This is a basic website based wallet, but it supports WalletConnect. 

**Marble (Position 18)** = `Does not work`
This is a very basic website based wallet, it has WalletConnect on its platform. However, it has a big issue that it does not actually work at all with WalletConnect even though it shows that it supports WalletConnect. I have tried v1 and v2 WalletConnect and neither work meaning either they have not finished integrating WalletConnect, Removed Support or it was wrongly integrated and simply just does not work! 

**Crossmint (Position 19)** = `Does not work`
This is a basic website based wallet, it has WalletConnect on its platform. However, it has a big issue that it does not actually work at all with WalletConnect even though it shows that it supports WalletConnect. I have tried v1 and v2 WalletConnect and neither work meaning either they have not finished integrating WalletConnect, Removed Support or it was wrongly integrated and simply just does not work! 

**Pier (Position 20)** = `Needs removed`
This give as 404 page as website does not seem to exist anymore. I was able to find a different website by searching on Google and found pierwallet.com instead of pierwallet.xyz that WalletConnect goes to, however the pierwallet.com shows the wallet is not on desktop and only available on mobile for Android and iOS.

Website to check: https://www.pierwallet.com/

**Paper (Position 21)** = `Does not work`
The wallet is a web wallet which only allows creating a wallet via email, however on creating a wallet it always fails to create making it impossible to test if they support WalletConnect or not!

Website to check: https://withpaper.com/wallet

**Rice Wallet (Position 22)** = `Needs removed `
On clicking to connect this give a 404 page error, however I checked and they dont have a desktop wallet. They only have a mobile wallet for android and iOS.

Website to check: https://ricewallet.io/ 

**Bobablocks (Position 23)** = `Needs removed` 
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://www.bobablocks.io/

**KryptoGo (Position 24)** = `Needs removed `
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://www.kryptogo.com/products/wallet

**Wallet3 (Position 25)** = `Needs removed `
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://wallet3.io/

**Prema (Position 26)** = `Needs removed `
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://premanft.com/en/home-en/

**OneKey (Position 27)** = `Unable to check`
They have a desktop wallet, however I was unable to find WalletConnect on their Desktop wallet.

**SecuX (Position 28)** = `Needs removed `
On clicking to connect this give a 404 page error, however I checked and they dont have a desktop wallet. They only have a mobile wallet for android and iOS.

Website to check: https://wsweb.secuxtech.com/

**Venly (Position 29)** = `Does not work`
They have a website based wallet, but it seems to be mostly for developers to test their social login rather then users. As it doesn't work, doesn't show assets, balances, can't send or receive and the only thing that can be done is sign messages from WalletConnect, however that does not seem to work and constantly gives errors.

Website to check: https://walletconnect.venly.io/

**Bitski (Position 30)** = `Unable to test`
I tried Bitski, however it requires an email to create a wallet and on creating it never sends an email to confirm the wallet so was unable to actually test this. at all

**SafeMoon (Position 31)** = `Needs removed `
This is a mobile only wallet which is available only on Android & iOS. It does not have a desktop wallet!

Website to check: https://safemoon.com/products#products_wallet

### Overview
I have noticed for months there has been issues with the WalletConnect list having a lot of mobile wallets or DApps on the desktop wallet list, however with this latest update last week it has now made the desktop WalletConnect list more of a mess. It is making use of WalletConnect for users terrible as they click to connect with a desktop wallet or use a desktop wallet and only find DApps or mobile wallets rather then Desktop wallets. It's also a massive pain for us developers as users then reach out to us on discord or support asking which wallet to use as none work on desktop or don't seem to work.

Suggestion to resolve this:
1- Remove all none desktop wallets which is a must, this includes all mobile and DApps that are listed as desktop wallets.

2- The ranking of desktop wallets has some how been changed and should be placed back to how it has been for years, with all the latest desktop wallets following on from there:
```
1- Ledger
2- Infinity Wallet
3- Ambire 
4- Now Wallet
...
```

I spent around 2-3 hours testing every wallet, visiting all websites to collect all the information above. I hope that it can be useful in solving this serious issue that WalletConnect currently is having with the Desktop Wallet list.

**Comment by crypblizz8** [comment]
Hey @Crytogenz, really appreciate the in-depth testing and reviewing on this team. 

I have passed this onto our team for review and changes. Will keep you posted and once again appreciate your time and help on this 🤝

**Reply by Crytogenz** [reply]
Hi @crypblizz8, glad to help and look forward to it been fixed.

**Comment by Crytogenz** [comment]
Hi @crypblizz8, there seems to be more new wallets added which are either DApps or only mobile. Do you guys check projects applying to be on the list? As it seems currently like any DApp can say its a wallet and be on the wallet list.

Personally as a dev I don't want to list DApps on my DApp when WalletConnect is supposed to list wallets. There is quite a few DApps from Zerion, CoinStats and more on the list.

Some newly added projects that are not desktop wallets to add to the list above include:

Me Wallet - `Needs removed`
This has a web wallet but it does not have WalletConnect support at all, it should not be on the desktop wallet list. It has a mobile wallet though. So needs removed only from desktop list as there is no desktop walletconnect support.

Stickey - `Needs removed`
This wallet the walletconnect on click goes to an invalid domain so not implemented correctly, also I checked their website and they don't even have a desktop wallet only android and iOS as can be checked here https://stickey.app/ 

CoinStats = `Needs removed as DApp`
This is a DApp and not a wallet, its a wallet DApp which basically requires a real wallet to connect with it. It uses WalletConnect the same as DApps to connect a real wallet to it.

**Comment by Crytogenz** [comment]
Hi @crypblizz8, was wondering if there was any update regarding the above? There seems to be a lot more issues now. Why is there no screening of projects to ensure they meet the requirements? I mean I would happily do it if needed as it should take less then 2 hours to fix a serious issue with WalletConnect at the moment

**Reply by glitch-txs** [reply]
hi @Crytogenz, for v2 I'd recommend taking a look at the[ `explorerRecommendedWalletIds / explorerExcludedWalletIds` options ](https://docs.walletconnect.com/2.0/web/walletConnectModal/modal/options#explorerrecommendedwalletids-optional). The team is currently working on Web3Modal v3 that's going to be more flexible on this.

**Comment by ruben-certhis** [comment]
@Cryptogenz Hi, Certhis is also a web wallet that uses email authentication.
